### PR TITLE
htseq/count module is implemented

### PR DESCRIPTION
**Pull request by zehrahazalsezer**
## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/htseq/count/meta.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        Groovy Map containing sample information
+        e.g. `[ id:'test', single_end:false ]`
+  - bam:
+      type: file
+      description: Sorted BAM/CRAM/SAM file
+      pattern: "*.{bam,cram,sam}"
+  - bai:
+      type: file
+      description: Contains indexed bam file
+      pattern: "*bam.bai"
```
Comment (kubranarci) [inline modules/nf-core/htseq/count/meta.yml:30]: .bai

**Review by kubranarci — COMMENTED** [review]
looks good to me

File: modules/nf-core/htseq/count/main.nf [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    label 'process_medium'
+
+    conda "bioconda::htseq=2.0.2"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/htseq:2.0.2--py310ha14a713_0':
+        'biocontainers/htseq:2.0.2--py310ha14a713_0' }"
+
+    input:
+    tuple val(meta), path(bam), path(bai)
+    tuple val(meta1), path(gtf)
```
Comment (nvnieuwk) [inline modules/nf-core/htseq/count/main.nf:14]:
```suggestion
    tuple val(meta2), path(gtf)
```
All modules use meta2 for the second meta map (just so you stay consistent with the other modules)

File: modules/nf-core/htseq/count/meta.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        Groovy Map containing sample information
+        e.g. `[ id:'test', single_end:false ]`
+  - bam:
+      type: file
+      description: Sorted BAM/CRAM/SAM file
+      pattern: "*.{bam,cram,sam}"
+  - bai:
+      type: file
+      description: Contains indexed bam file
+      pattern: "*.bai"
```
Comment (nvnieuwk) [inline modules/nf-core/htseq/count/meta.yml:30]: How about the CRAM index? Or is it not possible to supply it here?

File: modules/nf-core/htseq/count/meta.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      documentation: "https://htseq.readthedocs.io/en/latest/index.html"
+      doi: "10.1093/bioinformatics/btu638"
+      licence: "['GPL v3']"
+
+input:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. `[ id:'test', single_end:false ]`
```
Comment (nvnieuwk) [inline modules/nf-core/htseq/count/meta.yml:22]: Can you also add `meta2` to the input?

File: tests/modules/nf-core/htseq/count/nextflow.config [hunk 5]
```diff
@@ -0,0 +1,6 @@
+process {
+
+    publishDir = { "${params.outdir}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" }
+    docker.registry = 'quay.io'
```
Comment (nvnieuwk) [inline tests/modules/nf-core/htseq/count/nextflow.config:4]:
```suggestion
```

This already gets specified by the main config of the modules repository :)

File: tests/modules/nf-core/htseq/count/main.nf [hunk 6]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { HTSEQ_COUNT } from '../../../../../modules/nf-core/htseq/count/main.nf'
+
+workflow test_htseq_count {
```
Comment (nvnieuwk) [inline tests/modules/nf-core/htseq/count/main.nf:7]: Can you also add a test with CRAM files?

**Review by nvnieuwk — COMMENTED** [review]
Hi, very well done already! I have some small comments here

**Review by nvnieuwk — APPROVED** [review]
Awesome! LGTM!

**Review by kubranarci — APPROVED** [review]

**Comment by nvnieuwk** [comment]
Something seems wrong with your `versions.yml` file can you have a look at that? (possibly some invalid character or missing process name)

File: modules/nf-core/htseq/count/main.nf [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    htseq-count \\
+        ${input} \\
+        ${gtf} \\
+        ${args} \\
+        > ${prefix}.txt
+
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        : \$(echo \$(htseq-count --version ) | sed 's/^.*htseq-count //; s/Using.*\$//' ))
```
Comment (nvnieuwk) [inline modules/nf-core/htseq/count/main.nf:35]:
```suggestion
        htseq: \$(echo \$(htseq-count --version ) | sed 's/^.*htseq-count //; s/Using.*\$//' ))
```

File: modules/nf-core/htseq/count/main.nf [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    touch ${prefix}.txt
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        : \$(echo \$(htseq-count --version ) | sed 's/^.*htseq-count //; s/Using.*\$//' ))
```
Comment (nvnieuwk) [inline modules/nf-core/htseq/count/main.nf:46]:
```suggestion
        htseq: \$(echo \$(htseq-count --version ) | sed 's/^.*htseq-count //; s/Using.*\$//' ))
```

**Review by nvnieuwk — COMMENTED** [review]
This should fix it!

**Pull request merged**